src/Utilities/model.jl

```diff
@@ -416,7 +416,10 @@ function MOI.supports(
         MOI.ScalarQuadraticFunction{T}}}) where T
     return true
 end
-function MOI.set(model::AbstractModel, ::MOI.ObjectiveFunction, f::MOI.AbstractFunction)
+function MOI.set(model::AbstractModel, attr::MOI.ObjectiveFunction, f::MOI.AbstractFunction)
+    if !MOI.supports(model, attr)
+        throw(MOI.UnsupportedAttribute(attr))
+    end
```

Comment:
@blegat how breaking is this? It turns out, even if you said you didn't support it, it would still accept it.